bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/image/ImageCacheWatcher.scala

```diff
@@ -0,0 +1,124 @@
+package edu.gemini.catalog.image
+
+import java.nio.file._
+import java.nio.file.StandardWatchEventKinds._
+import java.nio.file.attribute.BasicFileAttributes
+import java.time.Instant
+import java.util.concurrent.Executors
+import java.util.logging.{Level, Logger}
+
+import jsky.util.Preferences
+
+import scalaz.concurrent.Task
+import scala.collection.JavaConverters._
+import scalaz._
+import Scalaz._
+
+/**
+  * Observes the file system to keep the cache populated
+  */
+object ImageCacheWatcher {
+  val Log: Logger = Logger.getLogger(this.getClass.getCanonicalName)
+
+  /**
+    * Populates the cache when the application starts
+    */
+  private def populateInitialCache(cacheDir: Path): Task[StoredImages] = {
+    /**
+      * Creates a stream of files in a dir
+      */
+    def initStream: Task[DirectoryStream[Path]] =
+      Task.delay(Files.newDirectoryStream(cacheDir, "img_*"))
+
+    /**
+      * Read each file and its access time to populate the cache
+      */
+    def readFiles(stream: DirectoryStream[Path]): Task[StoredImages] = {
+      // This will read the last access time of the file. Note that this is OS dependent,
+      // it may be disabled in some systems and we'd get creation time instead
+      def lastAccessTime(file: Path): Instant =
+        Files.readAttributes(file, classOf[BasicFileAttributes]).lastAccessTime.toInstant
+
+      def fileAndTime(f: Path): Option[Task[StoredImages]] =
+        ImageInFile.entryFromFile(f.toFile).map(StoredImagesCache.addAt(lastAccessTime(f), _))
+
+      val populateCache =
+        for {
+          cacheFiles  <- Task.delay(stream.iterator().asScala.toList)
+          accessTimes <- cacheFiles.flatMap(fileAndTime).sequenceU
+        } yield accessTimes
+      populateCache *> StoredImagesCache.get
+    }
+
+    /**
+      * Delete temporary files. This is done at startup, there maybe temporal
+      * files left in the cache if e.g. the OT exits in the middle of a download
+      */
+    def rmTempFiles: Task[Path] = Task.delay {
+      val matcher = FileSystems.getDefault.getPathMatcher("glob:.img*.fits")
+
+      Files.walkFileTree(cacheDir, new SimpleFileVisitor[Path] {
+        override def visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult = {
+          super.visitFile(file, attrs)
+          if (matcher.matches(file.getFileName)) {
+            file.toFile.delete()
+          }
+          FileVisitResult.CONTINUE
+        }
+      })
+    }
+
+    def closeStream(stream: DirectoryStream[Path]): Task[Unit] = Task.delay(stream.close())
+
+    for {
+      _  <- rmTempFiles
+      ds <- initStream
+      ab <- readFiles(ds).onFinish(_ => closeStream(ds)) // Make sure the stream is closed
+    } yield ab
+  }
+
+  /**
+    * Observe the file system to detect when files are modified and update the cache accordingly
```

Comment:
This would detect if somebody modifies the files outside the OT, it is a bit overkill since we will check for deletions anyway before displaying in the TPE, At some moment I thought about hashing the files to detect modifications and the watcher would have been more useful then, anyway it doesn't hurt